crates/networking/p2p/sync.rs

```diff
@@ -436,18 +438,15 @@ impl SyncManager {
         // For more details, refer to the `get_block_hash` function in [`LevmDatabase`] and the [`revm::Database`].
         store
             .mark_chain_as_canonical(&blocks)
+            .await
             .map_err(SyncError::Store)?;
 
         // Executing blocks is a CPU heavy operation
         // Spawn a blocking task to not block the tokio runtime
-        let res: Result<(), (ChainError, Option<BatchBlockProcessingFailure>)> = {
+        let res = {
             let blockchain = self.blockchain.clone();
-            tokio::task::spawn_blocking(move || {
-                Self::add_blocks(blockchain, &blocks, sync_head_found)
-            })
-            .await
-            .map_err(SyncError::JoinHandle)
-        }?;
+            Self::add_blocks(blockchain, blocks, sync_head_found).await
```

Comment:
We might still need this `spawn_blocking` in case it hogs the CPU. But if that's the case, I believe the proper place is inside the call.